### ISSUE TITLE
[BUG] Enabling vlan on a bonded NIC breaks the Harvester setup

### ISSUE THREAD

**Issue by gitlawr** [harvester/harvester#1541]
**Describe the bug**

**To Reproduce**
1. Install a 3-node cluster using https://github.com/harvester/ipxe-examples/tree/main/vagrant-pxe-harvester
2. Enable vlan network using the bond0 NIC

**Result**
Harvester dashboard fails to load.
rke2-server service crashes.

**Environment:**
 - Harvester ISO version:  master-119c2c48-head

**Additional context**
1. journalctl -u rke2-server
```
Nov 13 08:17:28 harvester-node-0 rke2[5452]: time="2021-11-13T08:17:28Z" level=warning msg="not running in CIS mode"
Nov 13 08:17:28 harvester-node-0 rke2[5452]: time="2021-11-13T08:17:28Z" level=fatal msg="no default routes found in \"/proc/net/route\" or \"/proc/net/ipv6_route\""
Nov 13 08:17:28 harvester-node-0 systemd[1]: rke2-server.service: Main process exited, code=exited, status=1/FAILURE
Nov 13 08:17:28 harvester-node-0 systemd[1]: rke2-server.service: Failed with result 'exit-code'.
Nov 13 08:17:28 harvester-node-0 systemd[1]: rke2-server.service: Unit process 2465 (containerd-shim) remains running after unit stopped.
```
2. The default route is deleted and IP is not assigned to harvester-br0
```
# $ ip a
5: bond0: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue master harvester-br0 state UP group default qlen 1000
    link/ether 52:54:00:b2:f8:b9 brd ff:ff:ff:ff:ff:ff
    inet 192.168.121.44/24 brd 192.168.121.255 scope global bond0
       valid_lft forever preferred_lft forever
    inet6 fe80::5054:ff:feb2:f8b9/64 scope link
       valid_lft forever preferred_lft forever
74: harvester-br0: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 52:54:00:b2:f8:b9 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::80d2:d1ff:fe0d:7ae2/64 scope link
       valid_lft forever preferred_lft forever
```
3. network controller logs
```
I1113 07:51:33.004717       1 main.go:110] Starting network controller with 2 threads.
I1113 07:51:33.004759       1 main.go:113] Starting network controller with no namespace.
W1113 07:51:33.005084       1 client_config.go:614] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-11-13T07:51:37Z" level=info msg="Do not update the existing CRD NodeNetwork"
time="2021-11-13T07:51:38Z" level=info msg="Do not update the existing CRD NetworkAttachmentDefinition"
I1113 07:51:40.801801       1 monitor.go:52] Start Monitor
time="2021-11-13T07:51:41Z" level=info msg="Starting k8s.cni.cncf.io/v1, Kind=NetworkAttachmentDefinition controller"
time="2021-11-13T07:51:41Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=NodeNetwork controller"
I1113 07:51:41.311041       1 controller.go:95] node network configuration harvester-node-1-vlan has been changed, spec: {Description: NodeName:harvester-node-1 Type:vlan NetworkInterface:}
I1113 07:51:41.311203       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
E1113 07:51:41.311423       1 controller.go:326] Can not get VLAN, error: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1113 07:51:41.401794       1 controller.go:95] node network configuration harvester-node-1-vlan has been changed, spec: {Description: NodeName:harvester-node-1 Type:vlan NetworkInterface:}
I1113 07:51:41.401918       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
E1113 07:51:41.402102       1 controller.go:326] Can not get VLAN, error: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1113 08:11:13.822730       1 controller.go:95] node network configuration harvester-node-1-vlan has been changed, spec: {Description: NodeName:harvester-node-1 Type:vlan NetworkInterface:bond0}
I1113 08:11:13.822888       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1113 08:11:13.822898       1 vlan.go:85] start to setup VLAN network
I1113 08:11:14.001396       1 link.go:326] delete route {Ifindex: 5 Dst: <nil> Src: <nil> Gw: 192.168.121.1 Flags: [] Table: 254}
I1113 08:11:14.001486       1 link.go:326] delete route {Ifindex: 5 Dst: 192.168.121.0/24 Src: 192.168.121.160 Gw: <nil> Flags: [] Table: 254}
I1113 08:11:14.001503       1 vlan.go:120] setup VLAN network successfully
```

**Comment by gitlawr** [comment]
To recover, get the IP of bond0 and run the following commands on each node:
```
ip a add <ip-of-bond0>/24 brd 192.168.121.255 dev harvester-br0
ip r add default via 192.168.121.1 dev harvester-br0
```

**Comment by TachunLin** [comment]
The recover command can restore harvester back to operation and VM with vlan network can get ip address
List out the steps for reference. 

1. Install 3 nodes harvester from vagrant ipxe-example master
2. Enable vlan on bond0
3. Now harvester dashboard page will out of work
4. We can't ssh to original assigned node ip
5. We need to ssh to 192.168.0.30, 31, 32  instead
6. Run ip a | grep bond0  get the {ip-of-bond0}
7. Run the recover command
```
ip a add <ip-of-bond0>/24 brd 192.168.121.255 dev harvester-br0
ip r add default via 192.168.121.1 dev harvester-br0
```
8. Confirm default route by using `ip r | grep default`
9. Wait for a while and access dashboard (it takes some time)
10. Create virtual machine and assign vlan network, confirm can get ip address

**Comment by yaocw2020** [comment]
> To recover, get the IP of bond0 and run the following commands on each node:
> 
> ```
> ip a add <ip-of-bond0>/24 brd 192.168.121.255 dev harvester-br0
> ip r add default via 192.168.121.1 dev harvester-br0
> ```

The harvster network controller will delete the route of the interface which is set for the VLAN network if it is not `harvester-mgmt`.
Why do you use the bond0 as the default route device? It should default to harvester-mgmt. 
If we don't take `harvester-mgmt` as the default route device, there may be something wrong with the management network. @TachunLin @gitlawr

**Comment by yaocw2020** [comment]
I wonder why we expose the default route configuration to users? @weihanglo 
https://github.com/harvester/harvester-installer/commit/f3107af8582b1caf6ca44dcbe542f8958288a7ff#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5

**Comment by bk201** [comment]
> I wonder why we expose the default route configuration to users? @weihanglo [harvester/harvester-installer@f3107af#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5](https://github.com/harvester/harvester-installer/commit/f3107af8582b1caf6ca44dcbe542f8958288a7ff#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5)

Because the user might have a preferred gateway on other networks.

**Comment by TachunLin** [comment]
@yaocw2020 

The reason we use the default `bond0` as the default route device is due to while we are trying to set up an airgapped rancher to connect to online harvester for testing issue #1247 (Provision RKE2 cluster from airgapped rancher)   
We setup the environment by using vagrant ipxe example, but found if we use default `harvester-mgmt` as the default route, we can't retrieve IP for rke2 cluster vm. Thus we try to change it to `bond0` and cause harvester network not working.  
If any user try to enable cluster network in `bond0` they may encounter the same issue. 
I think that is why we have this issue filed and recover command for tracking.

**Comment by yaocw2020** [comment]
> > I wonder why we expose the default route configuration to users? @weihanglo [harvester/harvester-installer@f3107af#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5](https://github.com/harvester/harvester-installer/commit/f3107af8582b1caf6ca44dcbe542f8958288a7ff#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5)
> 
> Because the user might have a preferred gateway on other networks.

We should keep the default gateway interface stable to keep the management network stable. Users may set the route via the preferred gateway towards the target network.

**Comment by TachunLin** [comment]
Verified **fixed** on master-02e88c00-head and the latest [ipxe-examples/pull/32](https://github.com/harvester/ipxe-examples/pull/32)

### Result
Enable virtual network with `bond0` will not make harvester service out of work. 
![image](https://user-images.githubusercontent.com/29251855/143804059-f8fc0bee-b42a-4daa-b0bb-438b64b75db2.png)

Host can set to `bond` accordinly
![image](https://user-images.githubusercontent.com/29251855/143803958-892e77ed-8dad-43f9-bb68-609b0efa8019.png)

### Test Information
* Test Environment: 3 nodes harvester on local kvm machines
* Harvester version: master-02e88c00-head

### Verify Steps

1. Get [ipxe-examples/pull/32](https://github.com/harvester/ipxe-examples/pull/32)
2. Vagrant pxe install 3 nodes harvester
3. Access harvester settings page
4. Open `settings` -> `vlan`
5. Enable virtual network and set with `bond0`
6. Navigate to every page to check harvester is working
7. Create a vlan based on bon0
